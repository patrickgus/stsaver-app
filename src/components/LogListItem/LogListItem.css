.LogListItem {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.LogListItem__buttons button {
  width: 80px;
  margin-right: 10px;
}
