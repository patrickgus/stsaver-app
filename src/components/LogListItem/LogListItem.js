import React, { Component } from "react";
import { format } from "date-fns";
import { Input, Button } from "../Utils/Utils";
import LogApiService from "../../services/log-api-service";
import LogContext from "../../contexts/LogContext";
import "./LogListItem.css";

class LogListItem extends Component {
  constructor(props) {
    super(props);
    this.state = {
      editing: false
    };
  }

  static contextType = LogContext;

  handleSubmit = e => {
    e.preventDefault();
    const { userId } = this.context;
    const { start_time, end_time, media, breaks } = e.target;

    LogApiService.updateLog(
      userId,
      this.props.log.id,
      start_time.value,
      end_time.value,
      media.value,
      breaks.value
    ).then(res => {
      LogApiService.getLogs(userId).then(this.context.setLogList);
      this.setState({ editing: false })
    });
  };

  handleCancel = e => {
    e.preventDefault();
    this.setState({ editing: false });
  };

  handleEdit = () => {
    this.setState({ editing: true });
  };

  handleDelete = () => {
    const { userId } = this.context;
    LogApiService.deleteLog(userId, this.props.log.id).then(res => {
      LogApiService.getLogs(userId).then(this.context.setLogList);
    });
  };

  renderEditingView = () => {
    const log = this.props.log;
    const start_time = format(new Date(log.start_time), "yyyy-MM-dd'T'HH:mm");
    const end_time = format(new Date(log.end_time), "yyyy-MM-dd'T'HH:mm");

    return (
      <form className="LogListItem__edit-form" onSubmit={this.handleSubmit}>
        <div className="LogListItem__form-section">
          <label htmlFor="start_time">Start time</label>
          <Input
            type="datetime-local"
            name="start_time"
            id="start_time"
            defaultValue={start_time}
            required
          />
        </div>
        <div className="LogListItem__form-section">
          <label htmlFor="end_time">End time</label>
          <Input
            type="datetime-local"
            name="end_time"
            id="end_time"
            defaultValue={end_time}
            required
          />
        </div>
        <div className="LogListItem__form-section">
          <label htmlFor="media">Media type</label>
          <select name="media" id="media" defaultValue={log.media} required>
            <option value="">--Choose one--</option>
            <option value="computer">Computer</option>
            <option value="phone">Phone</option>
            <option value="tablet">Tablet</option>
            <option value="television">Television</option>
          </select>
        </div>
        <div className="LogListItem__form-section">
          <label htmlFor="breaks">Number of breaks</label>
          <Input
            type="number"
            name="breaks"
            id="breaks"
            defaultValue={log.breaks}
            required
          />
        </div>
        <div className="LogListItem__buttons">
          <Button type="submit">Submit</Button>
          <Button onClick={this.handleCancel}>Cancel</Button>
        </div>
      </form>
    );
  };

  renderDefaultView = () => {
    const log = this.props.log;
    // console.log(log.id, log.start_time);
    // console.log(log.id, new Date(log.start_time));
    
    return (
      <>
        <h3>{format(log.start_time, "MM-dd-yyyy")}</h3>
        <p>
          {format(log.start_time, "h:mma")} -{" "}
          {format(log.end_time, "h:mma")}
        </p>
        <dl>
          <dt>Screen Hours:</dt>
          <dd>{log.hours}</dd>
          <dt>Media type:</dt>
          <dd>{log.media}</dd>
          <dt>Breaks taken:</dt>
          <dd>{log.breaks}</dd>
        </dl>
        <div className="LogListItem__buttons">
          <Button className="edit" onClick={this.handleEdit}>Edit</Button>
          <Button onClick={this.handleDelete}>Delete</Button>
        </div>
      </>
    );
  };

  render() {
    return (
      <section className="LogListItem">
        {this.state.editing
          ? this.renderEditingView()
          : this.renderDefaultView()}
      </section>
    );
  }
}

export default LogListItem;
