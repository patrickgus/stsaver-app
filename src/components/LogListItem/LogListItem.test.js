import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import LogListItem from "./LogListItem";
import logs from "../../stores/logs-store";

describe.only("LogListItem component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <LogListItem log={logs[0]} />
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders the item as expected", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <LogListItem log={logs[0]} />
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
