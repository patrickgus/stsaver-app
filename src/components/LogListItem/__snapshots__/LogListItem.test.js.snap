// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogListItem component renders the item as expected 1`] = `
<section
  className="LogListItem"
>
  <h3>
    03-12-2020
  </h3>
  <p>
    4:00PM
     -
     
    8:00PM
  </p>
  <dl>
    <dt>
      Screen Hours:
    </dt>
    <dd />
    <dt>
      Media type:
    </dt>
    <dd>
      television
    </dd>
    <dt>
      Breaks taken:
    </dt>
    <dd>
      0
    </dd>
  </dl>
  <div
    className="LogListItem__buttons"
  >
    <button
      className="Button edit"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="Button "
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</section>
`;
